File: solutions/TypeScript/[0485_Easy] Max Consecutive Ones/findMaxConsecutiveOnes.ts
function findMaxConsecutiveOnes(nums: number[]): number {
  let max = 0;
  let count = 0;

  for (const n of nums) {
    if (n === 1) {
      count++;
    } else {
      if (count > max) {
        max = count;
      }
      count = 0;
    }
  }

  return count > max ? count : max;
}

File: solutions/Java/[0485_Easy] Max Consecutive Ones/findMaxConsecutiveOnes.java
class Solution {
  public int findMaxConsecutiveOnes(int[] nums) {
    int max = 0;
    int count = 0;

    for (int n: nums) {
      if (n == 0) {
        max = Math.max(max, count);
        count = 0;
        continue;
      }

      count++;
    }

    return Math.max(max, count);
  }
}
